### ISSUE TITLE
Action should not be re-executed when updating to provider to >= 1.13.0

### ISSUE THREAD

**Issue by royveshovda** [Azure/terraform-provider-azapi#482]
I got the same issue as described in https://github.com/Azure/terraform-provider-azapi/issues/411. The workaround described was all good for versions 1.11.0-1.12.1. However, this workaround stopped working when upgrading to 1.13.0. I have tried both

```
lifecycle {
  ignore_changes = [when]
}

```
and

```
lifecycle {
  ignore_changes = all
}
```

The plan always shows it will update the resource and append `when="apply"`, and unfortunately the API call I am making is not idempotent, and apply fails (yes, I know it is fragile, but we are talking a PIM-related call, which is something I will restrain myself from describing here 😅).

I do not understand the root cause, I am afraid. But these are my experiences. Happy to test different workarounds, if you have any suggestions?

**Comment by ms-henglu** [comment]
Hi @royveshovda ,

Thank you for taking time to report this issue!

I'll open a PR shortly to fix it. I can think of some workarounds:
1. Downgrade to 1.12.1. (recommended)
2. Modify the terraform state and set "when=apply" to the `azapi_resource_action`. But it's not recommend to modify the terraform state directly.

**Comment by royveshovda** [comment]
Thanks, @ms-henglu .

I opened the state file and considered to modify.....decided to go for workaround 1) 😅

**Comment by ms-henglu** [comment]
Hi @royveshovda ,

The bug fix has been released in v1.13.1, would you please have a try? Thanks.

**Comment by royveshovda** [comment]
I can confirm, @ms-henglu , that with version 1.13.1 it works for our workflow. Thanks a lot.
(not sure how you prefer, but I closed this issue)
